validate/secretsFile.go

```diff
@@ -0,0 +1,48 @@
+package validate
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/xeipuuv/gojsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+// Validate secrets file and returns true or false depending on the validation result.
+// Outputs error message if validation fails including what the issue is.
+// Debug message is logged if debug is true and validation succeeded.
+func SecretsFile(fileToValidate string, schemaPath string) bool {
+	documentFile, err := ioutil.ReadFile(fileToValidate)
+	if err != nil {
+		panic(err)
+	}
+
+	y, err := yaml.YAMLToJSON(documentFile)
```

Comment:
The description of the function `YAMLToJSON` is
> YAMLToJSON converts YAML to JSON. Since JSON is a subset of YAML, passing JSON through this method should be a no-op.

And I'm implementing some json tests as well. It works very well.

Reply:
I've made json tests now as well and also changed so the inline secret "file" thing will be validated too.